api/config/routes.rb

```diff
@@ -133,6 +133,16 @@
       end
     end
 
+    resources :address_book, only: [], param: :address_id do
+      collection do
+        get :mine
+        post :update
+      end
+      member do
+        post :remove
+      end
+    end
```

Comment:
I see the argument of "the address book itself is the resource." But if it's taking all of us a while to wrap our heads around, then maybe that's just not worth it? It wasn't intuitive to me either. What if we went with scoping it in the user? I.e. `/api/users/34/addresses`, `/api/users/34/addresses/1234`? That would have the added advantage of allowing you to build an admin panel around that API that manages others' addresses (though then we'd have to check for that permission). That seems more intuitive to me.